### PR TITLE
Fullsim

### PR DESCRIPTION
**Pull request by sanghwapark**
Adding full simulation analysis module

### CODE REVIEW

File: src/Analysis.cxx [hunk 1]
```diff
@@ -453,7 +453,7 @@ void Analysis::Execute() {
 	  wJetTotal += wJet;
 
       for(int i = 0; i < kin->jetsRec.size(); i++){
-        PseudoJet jet = kin->jetsRec[i];
+	fastjet::PseudoJet jet = kin->jetsRec[i];
```
Comment (c-dilks) [inline src/Analysis.cxx:456]: I will add `fastjet::` in the centauro loop too

File: src/Kinematics.cxx [hunk 2]
```diff
@@ -325,26 +325,26 @@ void Kinematics::GetJets(
...
   double R = 0.8;
-  JetDefinition jet_def(antikt_algorithm, R);
+  fastjet::JetDefinition jet_def(fastjet::antikt_algorithm, R);
 
-  csRec = ClusterSequence(particles, jet_def);
-  csTrue = ClusterSequence(particlesTrue, jet_def);
-  jetsRec = sorted_by_pt(csRec.inclusive_jets());
-  jetsTrue = sorted_by_pt(csTrue.inclusive_jets());
+  csRec = new fastjet::ClusterSequence(particles, jet_def);
+  csTrue = new fastjet::ClusterSequence(particlesTrue, jet_def);
```
[the remark this answers is not in the document]
Comment (c-dilks) [reply src/Kinematics.cxx]: resolved by 1878a01ab9decef1149b3760f40a1b7fc045c31f

File: src/AnalysisDD4hep.cxx [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+  TTreeReaderArray<float> EcalBarrelClusters_y(tr,      "EcalBarrelImagingClusters.position.y");
+  TTreeReaderArray<float> EcalBarrelClusters_z(tr,      "EcalBarrelImagingClusters.position.z");
+  TTreeReaderArray<float> EcalBarrelClusters_theta(tr,  "EcalBarrelImagingClustersInfo.polar.theta");
+  TTreeReaderArray<float>  EcalBarrelClusters_phi(tr,    "EcalBarrelImagingClustersInfo.polar.phi");
+
+  TTreeReader::EEntryStatus entrystats = tr.SetEntry(0);
+
+  // open output file
+  // FIXME: replace name variable
+  outFile = new TFile(outfilePrefix,"RECREATE");
```
Comment (c-dilks) [inline src/AnalysisDD4hep.cxx:133]: we can figure out how to better use inheritance between the two Analysis classes, so for example the output file name is created in one place; I can help with this when we refactor `histSet` (see https://github.com/c-dilks/largex-eic/pull/7)

File: src/AnalysisDD4hep.cxx [hunk 4]
```diff
@@ -0,0 +1,735 @@
...
+	  clus->x = EcalEndcapNClusters_x[icl]; // use edep?
+	  clus->y = EcalEndcapNClusters_y[icl]; // use edep?
+	  clus->z = EcalEndcapNClusters_z[icl]; // use edep?
+	  clus->theta = EcalEndcapNClusters_theta[icl]; // use edep?
+	  clus->phi = EcalEndcapNClusters_phi[icl]; // use edep?
+	  v_ecal_clusters.push_back(clus);
+	}
+
+      // calculate true DIS kinematics
+      kinTrue->CalculateDISbyElectron(); // generated (truth)
```
Comment (c-dilks) [inline src/AnalysisDD4hep.cxx:456]: should this be moved to be between the `mcparticles` loop and the calorimeters loops (L372)?

File: src/AnalysisDD4hep.cxx [hunk 5]
```diff
@@ -0,0 +1,735 @@
...
+	    // TODO: consider adding a `finalState` cut
+	    if( AN->writeSimpleTree && histSetFillList.size()>0 ) ST->FillTree(w);
+	
+	  }//if cut
+	}//trk loop
+
+      //Hadronic reconstruction method
+      kin->sigmah = (kin->vecElectron.E() - kin->vecElectron.Pz());
+      kin->Pxh = hpx - kin->vecElectron.Px();
+      kin->Pyh = hpy - kin->vecElectron.Py();
```
Comment (c-dilks) [inline src/AnalysisDD4hep.cxx:610]: Shouldn't these be set before calling one of the `Kinematics::CalculateDIS` methods? Looking closer in `Kinematics` it seems that `sigmah`, `Pxh`, and `Pyh` never get initialized! https://github.com/c-dilks/largex-eic/issues/22 is opened to take care of that

File: src/AnalysisDD4hep.cxx (hunk 4 above)
Comment (sanghwapark) [reply src/AnalysisDD4hep.cxx]: fixed by 23d4481

File: src/AnalysisDD4hep.cxx (hunk 5 above)
Comment (sanghwapark) [reply src/AnalysisDD4hep.cxx]: Moved final state loop before calling the DIS kinematics calculation. Fixed by the commit 23d4481.

File: src/Analysis.h [hunk 6]
```diff
@@ -51,6 +50,11 @@ class Analysis : public TNamed
 
     // add a new bin scheme
     void AddBinScheme(TString varname, TString vartitle);
+    void InitBinScheme();
```
[the remark this answers is not in the document]
Comment (sanghwapark) [reply src/Analysis.h]: Removed.

**Pull request merged**